docker-compose.yaml

```diff
@@ -232,4 +256,11 @@ services:
       - POSTGRES_PASSWORD=hivepassword
       - POSTGRES_DB=hive
     volumes:
-      - ./cdr/cdm/jupyter/cdm-postgres:/var/lib/postgresql/data  # For local development only. In Rancher development, PostgreSQL data shouldn't be stored in a shared mount.
\ No newline at end of file
+      - ./cdr/cdm/jupyter/cdm-postgres:/var/lib/postgresql/data  # For local development only. In Rancher development, PostgreSQL data shouldn't be stored in a shared mount.
+    networks:
+      - cdm-jupyterhub-network
+
+networks:
+  # Created by `docker network create cdm-jupyterhub-network`
+  cdm-jupyterhub-network:
+    external: true
```

Comment:
What does external do here?

Reply:
it indicates the network is created by `docker network create`. I wasn't able to get it working locally with docker compose created networks.

Reply:
What implications does that have?

Do you have to do something similar in Rancher?

Reply:
I am working with Devops on this.

Reply:
Ok, so it's not clear what's going to happen in rancher yet.

I still don't understand what the difference is between the two network modes in docker compose

Reply:
With the one docker compose created automatically, I couldn't get jupterhub working locally. Hostname is not resolved correctly which is what we see in Rancher now. I don't know I have the best answer for this question. Still trying to figure it out. But with this setting, at least I can get jupterhub working properly locally with docker compose.

Reply:
Ok, sounds like this is "well it works but I don't know why" situation.

The reason I was asking this is because I don't know how that would translate to Rancher and if it'd have any security implications

Reply:
yea. I think I will try to understand and how it works in Rancher and maybe come back and update docker compose to make it similar if not the same as Rancher.